plugin/storage/badger/factory_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package badger
+
+import (
+	"expvar"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	assert "github.com/stretchr/testify/require"
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/config"
+)
```

Comment:
I'm not entirely sure how would this help? The temp dir is different for each instance, thus call to another Close wouldn't help at all? Each call to ioutil.TempDir creates a random number which is in the directory name.

I made the change now in the Close() that even if f.store.Close() fails, we will still delete the tempdir first instead of just returning the error.

Reply:
sorry, I simply meant to add that line as a validation that the factory implements io.Closer, because it is how the Close method is used by the meta-factory, it tries to cast the objects to io.Closer. If this factory changes its Close method signature, e.g. to not return an error, the cast will fail and Close() won't be called.

Reply:
I found it to be a good practice to be explicit about interfaces you need to implement, rather than relying on indirect testing of the fact. I.e. this in the test file sets the expectation unambiguously:

```go
var _ io.Closer = new(myType)
```